Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,903 @@
+                    $DBForPostgreSQLFlexibleServerDetails += $DBForPostgreSQLFSResource  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                    @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                    @{N = 'ResourceName'; E = { $_.Name } }, 
+                    @{N = 'TLSVersion'; E = { $tlsparameterValue } },
+                    @{N = 'SecureTransportStatus'; E = { $sslparameterValue } }
+
+                }
+                catch {
+                    Write-Host "Error fetching Azure Database for PostgreSQL flexible server(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+        }
+                                                                
+    }
+    
+    $totalDBForPostgreSQLFS = ($DBForPostgreSQLFlexibleServerDetails | Measure-Object).Count
+
+    if ($totalDBForPostgreSQLFS -eq 0) {
+        Write-Host "No Azure Database for PostgreSQL flexible server(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+  
+    Write-Host "Found [$($totalDBForPostgreSQLFS)] Azure Database for PostgreSQL flexible server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Azure Database for PostgreSQL flexible server(s) for which server parameter require_secure_transport in not ON
+    $DBForPostgreSQLFSWithNonSecureTLSVersionEnabled = @()
+
+    Write-Host "Separating Azure Database for PostgreSQL flexible server(s) for which only secure TLS version is not set or secure transport communication is disabled..." -ForegroundColor $([Constants]::MessageType.Info)
+
+    $DBForPostgreSQLFlexibleServerDetails | ForEach-Object {
+        if (-not ( (CheckIfOnlySecureTLSVersionConfigured($_.TLSVersion)) -and (CheckIfSSLConfigured($_.SecureTransportStatus)))) {
+            $DBForPostgreSQLFSWithNonSecureTLSVersionEnabled += $_
+        }
+        else {
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+            $logResource.Add("ResourceName", ($_.ResourceName))	
+            $logResource.Add("Reason", "TLS version(s) and secure transport configured on this Resource.")    	
+            $logSkippedResources += $logResource	
+        }
+    }
+   
+    $totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled = ($DBForPostgreSQLFSWithNonSecureTLSVersionEnabled  | Measure-Object).Count
+
+    if ($totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled -eq 0) {
+        Write-Host "No Azure Database for PostgreSQL flexible server(s) found with non-secure TLS version enabled and secure transport communication is disabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
NIT: Message is not intuitive. **and** to be replaced by **or**.

Reply:
Fixed